reposerver/repository/repository.go

```diff
@@ -98,6 +98,37 @@ func NewService(metricsServer *metrics.MetricsServer, cache *reposervercache.Cac
 	}
 }
 
+// List a subset of the refs (currently, branches and tags) of a git repo
+func (s *Service) ListRefs(ctx context.Context, q *apiclient.ListRefsRequest) (*apiclient.Refs, error) {
+	gitClient, err := s.newClient(q.Repo)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO check cache
```

Comment:
The `ls-remote` is used if every other repo server operation without caching. In fact it is used to find cache key. So I think caching is not required here and we can remove this comment.